pandas/_libs/tslibs/resolution.pyx

```diff
@@ -475,16 +481,23 @@ class _FrequencyInferer(object):
     def rep_stamp(self):
         return Timestamp(self.values[0])
 
-    def month_position_check(self):
+    cdef month_position_check(self):
         # TODO: cythonize this, very slow
-        calendar_end = True
-        business_end = True
-        calendar_start = True
-        business_start = True
-
-        years = self.fields['Y']
-        months = self.fields['M']
-        days = self.fields['D']
+        cdef:
+            int32_t daysinmonth, y, m, d
+            bint calendar_end = True
+            bint business_end = True
+            bint calendar_start = True
+            bint business_start = True
+            bint cal
+            int32_t[:] years
```

Comment:
These are memoryviews, right? Any reason you are choosing these instead of the `ndarray` declarations throughout the rest of the code base?

Reply:
My understanding is that memoryviews are encouraged because they are lighter weight.

Reply:
@jbrockmendel : That is indeed correct.  I agree with this design choice.

Reply:
In general, one of the main problems with memoryviews is with read-only arrays (and for that reason we do/should not use them solely in certain algos), but I don't think that's relevant in this case

Reply:
you can make cdefs readonly when it makes sense FYI